src/coreclr/src/jit/importer.cpp

```diff
@@ -1550,7 +1550,7 @@ var_types Compiler::impNormStructType(CORINFO_CLASS_HANDLE structHnd, var_types*
         {
             unsigned originalSize = info.compCompHnd->getClassSize(structHnd);
 
-            if ((originalSize >= minSIMDStructBytes()) && (originalSize <= maxSIMDStructBytes()))
+            if ((originalSize >= minSIMDStructBytes()) && (originalSize <= largestEnregisterableStructSize()))
```

Comment:
These changes seem fairly subtle... would be easy enough to think the upper check should still be `maxSIMDStructBytes()`.

I think we may need to make it clearer why `maxSIMDStructBytes` and `largestEnregisterableStructSize` differ and which one should be called when.

Reply:
Looking at this it would make more sense to me to build a helper function which encapsulates the call to minSIMDStructBytes and largestEnregisterableStructSize. Something like `structSizeMightRepresentSIMDType(size_t structSize)` ?

Reply:
Yes, I think that might make the intention clearer.